tron/bin/ddbbackup.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+import argparse
+import logging
+import os
+from datetime import datetime
+
+import boto3
+
+from tron.config import manager
+from tron.config import schema
+
+# Default values for arguments
+DEFAULT_WORKING_DIR = '/var/lib/tron/'
+DEFAULT_CONF_PATH = 'config/'
+MAX_BACKUPS = 5
+log = logging.getLogger(__name__)
+
+
+def create_backup(ddb, table_name):
+    backup_name = '{}-{}'.format(table_name, datetime.now().strftime('%Y%m%d%H%M'))
+    log.debug("Creating backup:", backup_name)
+    response = ddb.create_backup(
+        TableName=table_name, BackupName=backup_name)
+    log.debug(response)
+
+
+def delete_old_backups(ddb, table_name, max_backups):
+    log.debug("Deleting old backups for table:", table_name)
+
+    backups = ddb.list_backups(TableName=table_name)
+
+    backup_count = len(backups['BackupSummaries'])
+    log.debug('Total backup count:', backup_count)
+
+    if backup_count <= max_backups:
+        print("No stale backups. Exiting.")
+        return
+
+    sorted_list = sorted(backups['BackupSummaries'],
+                         key=lambda k: k['BackupCreationDateTime'])
+
+    old_backups = sorted_list[:max_backups]
+
+    for backup in old_backups:
+        arn = backup['BackupArn']
+        log.debug("ARN to delete: " + arn)
+        deleted_arn = ddb.delete_backup(BackupArn=arn)
+        status = deleted_arn['BackupDescription']['BackupDetails']['BackupStatus']
+        log.debug("Status:", status)
+
+    return
+
+
+def parse_cli():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-w",
+        "--working-dir",
+        default=DEFAULT_WORKING_DIR,
+        help="Working directory for the Tron daemon, default %(default)s",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-path",
+        default=DEFAULT_CONF_PATH,
+        help="File path to the Tron configuration file",
+    )
+    parser.add_argument(
+        "-m",
+        "--max-backups",
+        default=MAX_BACKUPS,
+        help="The maximum number of backups",
+    )
+    args = parser.parse_args()
+    args.working_dir = os.path.abspath(args.working_dir)
+    args.config_path = os.path.join(
+        args.working_dir,
+        args.config_path,
+    )
+    return args
+
+
+def read_config(args):
+    return manager.ConfigManager(args.config_path).load().get_master().state_persistence
+
+
+def main():
+    # Fetch configs. Arguments are passed from systemctl. You can find them in loggging.conf
```

Comment:
Is it still true that the arguments come from systemctl / logging.conf?
Seems like they just come from the command in puppet, in which case, I don't think you need to add an extra explanation.

Reply:
Oh thx. I forgot to remove the comments.